### ISSUE TITLE
Recent Updates Broke amdgpu-plot

### ISSUE THREAD

**Issue by Ricks-Lab** [Ricks-Lab/amdgpu-utils#22]
I suspect it was the latest matplotlib update, but I am now seeing a similar issue to this one after upgrading to 18.04.2:   [12892](https://github.com/matplotlib/matplotlib/issues/12892)

I have made this change (not committed), but still have an issue:
``` # Plot GPUs
            plt.figure(v["figure_num"])
            v["ax1"].clear()
            v["ax1"].set_ylabel('Loading/Power/Temp', color='k', fontsize=10)
            for plot_item in ['loading', 'power_cap', 'power', 'temp']:
                if gc.plot_items[plot_item] == True:
                    print([xi[0] for xi in (ldf[ldf["Card#"].isin([k])].loc[:,['Time']]).values.tolist()])
                    print([yi[0] for yi in (ldf[ldf["Card#"].isin([k])].loc[:,[plot_item]]).values.tolist()])
                    print("color: ", gc.colors[plot_item])
                    v["ax1"].plot(kind='line',
                            x=[xi[0] for xi in (ldf[ldf["Card#"].isin([k])].loc[:,['Time']]).values.tolist()],
                            y=[yi[0] for yi in (ldf[ldf["Card#"].isin([k])].loc[:,[plot_item]]).values.tolist()],
                            color=gc.colors[plot_item], linewidth=0.5)
                    v["ax1"].text(x=ldf[ldf["Card#"].isin([k])].loc[:,['Time']].iloc[-1],
                            y=ldf[ldf["Card#"].isin([k])].loc[:,[plot_item]].iloc[-1],
                            s=str(int(ldf[ldf["Card#"].isin([k])].loc[:,[plot_item]].iloc[-1].values[0])),
                            bbox = dict(boxstyle="round,pad=0.2", facecolor=gc.colors[plot_item]), fontsize=6)

            v["ax2"].clear()
            v["ax2"].set_ylabel('MHz/mV', color='k', fontsize=10)
            #ylim_val = 1.1* ldf[ldf["Card#"].isin([k])].loc[:,['vddgfx', 'sclk_f', 'mclk_f']].max().max()
            for plot_item in ['vddgfx', 'sclk_f', 'mclk_f']:
                if gc.plot_items[plot_item] == True:
                    v["ax2"].plot(kind='line',
                            x=[xi[0] for xi in (ldf[ldf["Card#"].isin([k])].loc[:,['Time']]).values.tolist()],
                            y=[yi[0] for yi in (ldf[ldf["Card#"].isin([k])].loc[:,[plot_item]]).values.tolist()],
                            color=gc.colors[plot_item], linewidth=0.5)
                    v["ax2"].text(x=ldf[ldf["Card#"].isin([k])].loc[:,['Time']].iloc[-1],
                            y=ldf[ldf["Card#"].isin([k])].loc[:,[plot_item]].iloc[-1],
                            s=str(int(ldf[ldf["Card#"].isin([k])].loc[:,[plot_item]].iloc[-1].values[0])),
                            bbox = dict(boxstyle="round,pad=0.2", facecolor=gc.colors[plot_item]), fontsize=6)
```
Original code still on master [amdgpu-plot](https://github.com/Ricks-Lab/amdgpu-utils/blob/master/amdgpu-plot).  Looking for help:

- Can someone confirm that this is working before upgrade to 18.04.2 and if so, what version of matplotlib are you using?
- Is my method of specifying plot data from a dataframe not optimal?

**Comment by Ricks-Lab** [comment]
@csecht 
Can you let me know if your current version of *amdgpu-plot* still works?  I have done a clean install of 18.04.2 which results in kernel 4.18 and perhaps newer versions of other packages.  I now can't get plot to work anymore.  

After you verify your current version, can you try the latest on master and use:
```
./amdgpu-plot --about
```
to get the version of matplotlib, pandas, and numpy.

Thanks!

**Comment by csecht** [comment]
Yep, it's broke. The plot window gives current information, but doesn't draw the plot lines. Attached is the terminal output.
[plot terminal output.txt](https://github.com/Ricks-Lab/amdgpu-utils/files/3149151/plot.terminal.output.txt)
Here is the relevant  *--about* info:
```
....
Version:  v2.5.0
Maintainer:  RueiKe
Status:  Stable Release
matplotlib version:  2.1.1
pandas version:  0.24.2
numpy version:  1.16.2
```
I'm running Ubuntu 18.04.2, with the Linux 4.15.0-48-generic kernel.
I did an Ubuntu system maintenance update yesterday.

**Comment by Ricks-Lab** [comment]
> Yep, it's broke. The plot window gives current information, but doesn't draw the plot lines. Attached is the terminal output.
> [plot terminal output.txt](https://github.com/Ricks-Lab/amdgpu-utils/files/3149151/plot.terminal.output.txt)
> Here is the relevant _--about_ info:
> 
> ```
> ....
> Version:  v2.5.0
> Maintainer:  RueiKe
> Status:  Stable Release
> matplotlib version:  2.1.1
> pandas version:  0.24.2
> numpy version:  1.16.2
> ```
> 
> I'm running Ubuntu 18.04.2, with the Linux 4.15.0-48-generic kernel.
> I did an Ubuntu system maintenance update yesterday.

Did you try the official last release of amdgpu-utils?  Since your packages are older versions than mine, I expected it to work. Can you try the official  release just in case I introduced another issue in the debug modifications?

**Comment by Ricks-Lab** [comment]
Here are the versions for my system:
```
matplotlib version:  3.0.3
pandas version:  0.24.2
numpy version:  1.16.3
```
So it looks like it may not be related to matplotlib, but it would be good if you can try the latest official release.

**Comment by csecht** [comment]
Right, so from the amdgpu-utils-2.5.0 download (Feature Release), *amdgpu-plot* is fully working.
From the latest amdgpu-utils-master, lines are not plotted.

**Comment by csecht** [comment]
Right you are. The latest master plots well.

**Comment by Ricks-Lab** [comment]
> Right you are. The latest master plots well.
> […](#)

Thanks for confirming. This would indicate that the behavior of Matplotlib 2.x is different than 3.x. My next step will be to try to install a 2.x version to verify. Also, I hope to get someone to review the code to see if there is something cludgy in the way I am making the plot call with pandas df elements.

**Comment by Ricks-Lab** [comment]
@csecht 
I found a solution.  With the latest version of matplotlib, I need to convert the string date to a datetime object.  I have made this changed and verified on my system with matplotlib 3.X.  Can you try the latest on master to make sure it still works for matplotlib 2.x?  
Thanks!

**Comment by csecht** [comment]
Yes, the latest master still works on my system with matplotlib 2.1.1. Congrats!

**Comment by Ricks-Lab** [comment]
@csecht   Thanks for the confirmation!

Fix released as 2.5.1.
